Item 7002: Automated test fixes related to shared component domain designer refactors for shared code

cnathe opened the pull request:
#### Rationale
Changes were made in the shared components package to consolidate shared pieces of the various domain designer components. Those changes were pulled into platform and SM with the related PRs and these test fixes were needed as a result.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/195
* https://github.com/LabKey/platform/pull/997
* https://github.com/LabKey/sampleManagement/pull/218

#### Changes
* Issue 39938: Fix for domain designer text option handling of the "max text length" input and radio options
* Re-enable the test cases in ColumnResizeTest (related to fix for 38838)
* Add SampleSetTest test case for verifying error messages for reserved field check on create sample set
* Update SampleSetParentColumnTest test case for parent alias validation against domain fields to check on both create and update of sample set

cnathe committed:
Item 7002: Re-enabled ColumnResizeTest test cases after fix for issue 39938
- Domain field fix to revert max text length input to 'Unlimited' if user input is larger than 4000

cnathe committed:
Merge branch 'develop' into fb_smSampleTypeDesignerPolish

cnathe committed:
Item 7002: Fix for SM test failures related to domain designer save button text change within the app
- the "Save" button for the various domain designers now uses the following text in the apps: "Finish Creating ..." and "Finish Updating ..."

cnathe committed:
Item 7002: SM Test fixes round 2 for changes in polish story

labkey-danield commented on src/org/labkey/test/components/domain/DomainFormPanel.java:
```diff
@@ -217,6 +215,21 @@ public String getPanelTitle()
+        // It would be better to not return a raw WebElement but who knows what the future holds, different alerts
+        // may show different controls.
+        return BootstrapLocators.errorBanner.existsIn(getDriver()) ? BootstrapLocators.errorBanner.findElement(getDriver()) : null;
+    }
```
I think this should be ok for now, but a possible trip up is that the scope of the existsIn is the Driver which, I believe, is the entire page. So if there is an error anywhere on the page this will find the first one it comes across.

A backlog item for the test team is to revisit the shared test component for panels. When we revisit that we should look more closely on how we find the error banner associated with a panel.

cnathe replied:
Good to know. Thanks.

labkey-danield commented on src/org/labkey/test/components/domain/DomainFormPanel.java:
```diff
@@ -238,12 +248,12 @@ public String getPanelAlertText()
     public WebElement getPanelAlertWebElement()
     {
+        getWrapper().waitFor(()-> BootstrapLocators.infoBanner.existsIn(getDriver()),
+                "the error alert did not appear as expected", 1000);
```
Something to keep in mind about the waitFor call, when the wait expires it will throw an exception if there is a message parameter ("the error alert did not appear as expected.").
Adding this waitFor and message now means this function could throw when previously it did not.

cnathe replied:
Good to know. Thanks. Looks like the only usage to date was in the SMAssayCreateTest, which I'll verify passes.

labkey-danield commented on src/org/labkey/test/components/labkey/ui/samples/SampleTypeDesigner.java:
```diff
@@ -27,13 +27,20 @@
+        this(driver);
+        _useFinishButton = useFinishButton;
+    }
+
```
Did you want to change this one as well to not pass a parameter but put the smarts/logic in the getSaveButton?

cnathe replied:
Bad timing. I checked in this change right after you completed your review. I'll re-request review now that this change is in. Thanks.

cnathe committed:
Item 7002: SM test refactor for SampleTypeDesigner handling of save vs finish button text

cnathe committed:
Fix typo

labkey-tchad commented on src/org/labkey/test/components/labkey/ui/samples/SampleTypeDesigner.java:
```diff
@@ -208,6 +209,24 @@ public String getParentAliasSelectText(int index)
         protected final WebElement saveButton = Locator.button("Save").findWhenNeeded(this);
 
+        // the SM app uses alternate text for the sample type designer save buttons
+        protected final WebElement finishButton = Locator.buttonContainingText("Finish").findWhenNeeded(this);
```
You could make a Locator that would match either button.
```suggestion
        protected final WebElement saveButton = Locator.union(Locator.button("Save"),
                Locator.buttonContainingText("Finish")).findWhenNeeded(this);
```

cnathe replied:
Awesome, thanks @labkey-tchad. I don't think I've ever used the Locator.union. I'll give it a try.

cnathe committed:
Item 7002: SampleTypeDesigner fix to use Locator.XPathLocator.union for "Save" vs "Finish" button usages

cnathe committed:
Merge branch 'develop' into fb_smSampleTypeDesignerPolish

cnathe committed:
Item 7002: Add test case for sample set creation validation check regarding parent alias name vs field name uniqueness

cnathe committed:
Item 7002: Remove sample type creation special handling for reserved fields
- remove code that was uniqifying the field name
- add a test case in SampleSetTest
